bench/benchmarks/dds_update_bucket.js

```diff
@@ -0,0 +1,241 @@
+
+        var workerTile = getWorkerTile(coordinate, url);
+
+        var actor = {
+            send: function(action, params, sendCallback) {
+                setTimeout(function() {
+                    if (action === 'get icons') {
+                        getIcons(params, sendCallback);
+                    } else if (action === 'get glyphs') {
+                        getGlyphs(params, sendCallback);
+                    } else assert(false);
+                }, 0);
+            }
+        };
+
+        if (!workerTile.data) {
+            // preload data
+            getTile(url, function(err, response) {
+                if (err) throw err;
+                var data = new VT.VectorTile(new Protobuf(new Uint8Array(response)));
+
+                // Copy the property data from the vector tile features so we
+                // can use it in updateProperties() during the actual benchmark
+                // In the real use case, the updated property data would be
+                // provided by client code and it wouldn't be pre-tiled, so a
+                // custom source would have to figure out how to generate this
+                // tile-specific payload, probably by joining on an id or
+                // something.
+                workerTile.__propertyData = {};
+                for (var layerId in data.layers) {
+                    var layer = data.layers[layerId];
+                    workerTile.__propertyData[layerId] = [];
+                    for (var i = 0; i < layer.length; i++) {
+                        var properties = layer.feature(i).properties;
+                        workerTile.__propertyData[layerId].push(properties);
+                    }
+                }
+
+                workerTile.parse(data, layerFamilies, actor, null, function(err) {
+                    if (err) return callback(err);
+                    eachCallback();
+                });
+            });
+        } else {
+            workerTile.updateProperties(workerTile.__propertyData, layerFamilies, actor, function(err) {
+                if (err) return callback(err);
+                eachCallback();
+            });
+        }
+    }, function(err) {
+        var timeEnd = performance.now();
+        callback(err, timeEnd - timeStart);
+    });
+}
+
+function asyncTimesSeries(times, work, callback) {
+    if (times > 0) {
+        work(function(err) {
+            if (err) callback(err);
+            else asyncTimesSeries(times - 1, work, callback);
+        });
+    } else {
+        callback();
+    }
+}
+
+var createLayerFamiliesCacheKey;
+var createLayerFamiliesCacheValue;
+function createLayerFamilies(layers) {
+    if (layers !== createLayerFamiliesCacheKey) {
+        var worker = new Worker({addEventListener: function() {} });
+        worker['set layers'](layers);
+
+        createLayerFamiliesCacheKey = layers;
+        createLayerFamiliesCacheValue = worker.layerFamilies;
+    }
+    return createLayerFamiliesCacheValue;
+}
```

Comment:
There's a large amount of redundant code between these two benchmarks and the buffer benchmark. Can you abstract that out?